lib/go/test/tests/equals_test.go

```diff
@@ -0,0 +1,174 @@
+package tests
+
+import (
+	"equalstest"
+	"strconv"
+	"testing"
+)
+
+func TestEquals(t *testing.T) {
+	basicTgt, basicSrc := genBasicFoo(), genBasicFoo()
+	if !basicTgt.Equals(basicSrc) {
+		t.Fatal("BasicEqualsFoo.Equals() test failed")
```

Comment:
These test failures should be `t.Error` instead of `t.Fatal`. Fatal stops the test from running, so if someone made a mistake in the future and failed multiple cases inside this test, they will only see the first failure in the output.

Fatal should only be used when the following tests are depending on this one (e.g. used for nil checks).

Reply:
Thanks for suggestion